### PR TITLE
cryptsetup-generator: use "/proc/cmdline" as source when appropriate

### PR DESCRIPTION
**Pull request by jlebon**
Right now, we always say `/etc/crypttab` even if the source was fully
derived from the kargs.

Let's match what `systemd-fstab-generator` does and use `/proc/cmdline`
when that's the case.

### CODE REVIEW

**Comment by anitazha** [comment]
LGTM. Waiting for someone with more context to merge.

File: src/cryptsetup/cryptsetup-generator.c [hunk 1]
```diff
@@ -541,7 +541,7 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                 if (proc_cmdline_value_missing(key, value))
                         return 0;
 
-                d = get_crypto_device(startswith(value, "luks-") ? value+5 : value);
+                d = get_crypto_device(startswith(value, "luks-") ? value+strlen("luks-") : value);
```
Comment (poettering) [inline src/cryptsetup/cryptsetup-generator.c]: actually startswith() returns the pointer after the prefix. Hence, by using gcc's syntax extension that the 2nd/2rd ternary op can be ommitted, we can make this even perttier:

```c
d = get_crypto_device(startswith(value, "luks-") ?: value);
```

could you rework the patch like that pls?

**Review by poettering — CHANGES_REQUESTED** [review]

**Comment by poettering** [comment]
Looks great, but the other line you touched could be even better, see above!

File: src/cryptsetup/cryptsetup-generator.c (hunk 1 above)
Comment (jlebon) [reply src/cryptsetup/cryptsetup-generator.c]: Thanks, updated!

**Pull request merged**